In this project we choose set out to determine the best naeighbborhod in Pittsburgh. A factor we both agreed could be used to determine was saftey as saftey is of the upmost priority when chooisng somewhere to live.

We decided to use overall neighboorhood saftey as our final metric. The three data sets we choose represnet a neighboorhoods saftey.

1. crimes committed since 2024 in Allengheny county. Neighboorhoods with less crime are gennarlly safer.
2. Playgorunds in Allengheny county. Reprensents saftey for children and raising a family
3. Fires in Allengheny county since 2013. Represents saftey and qualtiy of housing.

The final metric combines normalized crime, fire, and playground counts into a single score where crime and fires increase the value, while playgrounds decrease it. Lower values indicate safer neighborhoods because they reflect less crime and fire activity and more playgrounds. 

List Of Neighboorhoods

In [9]:
import pandas as pd
neighborhoods = pd.read_csv("Neighboorhoods.csv")

neighborhood_list = sorted(
    neighborhoods["hood"].dropna().str.strip().unique()
)

print(neighborhood_list)

['Allegheny Center', 'Allegheny West', 'Allentown', 'Arlington', 'Arlington Heights', 'Banksville', 'Bedford Dwellings', 'Beechview', 'Beltzhoover', 'Bloomfield', 'Bluff', 'Bon Air', 'Brighton Heights', 'Brookline', 'California-Kirkbride', 'Carrick', 'Central Business District', 'Central Lawrenceville', 'Central Northside', 'Central Oakland', 'Chartiers City', 'Chateau', 'Crafton Heights', 'Crawford-Roberts', 'Duquesne Heights', 'East Allegheny', 'East Carnegie', 'East Hills', 'East Liberty', 'Elliott', 'Esplen', 'Fairywood', 'Fineview', 'Friendship', 'Garfield', 'Glen Hazel', 'Greenfield', 'Hays', 'Hazelwood', 'Highland Park', 'Homewood North', 'Homewood South', 'Homewood West', 'Knoxville', 'Larimer', 'Lincoln Place', 'Lincoln-Lemington-Belmar', 'Lower Lawrenceville', 'Manchester', 'Marshall-Shadeland', 'Middle Hill', 'Morningside', 'Mount Washington', 'Mt. Oliver', 'New Homestead', 'North Oakland', 'North Shore', 'Northview Heights', 'Oakwood', 'Overbrook', 'Perry North', 'Perry Sou

Number of crimes commited in each neighboorhood since 2024

In [10]:
crimes = pd.read_csv("CrimeRate.csv", low_memory=False)
crimes["Neighborhood"] = crimes["Neighborhood"].str.strip()

crime_counts = []
for hood in neighborhood_list:
    count = (crimes["Neighborhood"] == hood).sum()
    crime_counts.append((hood, count))

crime_count_list = pd.DataFrame(
    crime_counts,
    columns=["Neighborhood", "Crime_Count"]
)

Amount of playgrounds in each neighboorhood

In [11]:
playgrounds = pd.read_csv("Playground Equipment.csv", low_memory=False)
playgrounds["neighborhood"] = playgrounds["neighborhood"].str.strip()

playground_counts = []
for hood in neighborhood_list:
    count = (playgrounds["neighborhood"] == hood).sum()
    playground_counts.append((hood, count))

playgrounds_counts_list = pd.DataFrame(
    playground_counts,
    columns=["Neighborhood", "Playground_Count"]
)


Fires per neighborhood since 2013

In [12]:
fires = pd.read_csv("Fires.csv", low_memory=False)
fires["neighborhood"] = fires["neighborhood"].str.strip()

fire_counts = []
for hood in neighborhood_list:
    count = (fires["neighborhood"] == hood).sum()
    fire_counts.append((hood, count))

fires_counts_list = pd.DataFrame(
    fire_counts,
    columns=["Neighborhood", "Fire_Count"]
)


Creating a final metric

In [13]:
merged = crime_count_list.merge(fires_counts_list, on="Neighborhood")
merged = merged.merge(playgrounds_counts_list, on="Neighborhood")

merged["Crime_N"] = merged["Crime_Count"] / merged["Crime_Count"].max()
merged["Fire_N"] = merged["Fire_Count"] / merged["Fire_Count"].max()
merged["Play_N"] = merged["Playground_Count"] / merged["Playground_Count"].max()

merged["Metric"] = (
    merged["Crime_N"] +
    2 * merged["Fire_N"] -
    merged["Play_N"]
)

print(merged.sort_values("Metric").head())

        Neighborhood  Crime_Count  Fire_Count  Playground_Count   Crime_N  \
65     Regent Square          107           9                 9  0.011717   
81    Swisshelm Park           76          17                 6  0.008322   
58           Oakwood          122          13                 4  0.013360   
54     New Homestead           61          22                 5  0.006680   
0   Allegheny Center          683          63                15  0.074792   

      Fire_N   Play_N    Metric  
65  0.027439  0.28125 -0.214655  
81  0.051829  0.18750 -0.075519  
58  0.039634  0.12500 -0.032372  
54  0.067073  0.15625 -0.015424  
0   0.192073  0.46875 -0.009812  


5 safets neighboorhoods

In [14]:
for _, row in merged.sort_values("Metric").head(5).iterrows():
    print(row["Neighborhood"], "-", round(row["Metric"], 4))

Regent Square - -0.2147
Swisshelm Park - -0.0755
Oakwood - -0.0324
New Homestead - -0.0154
Allegheny Center - -0.0098


We found that the best neighboorhood based on our metric was by far Regent Square with a saftey score of -0.2147, the next closest being Swisshelm Park with a score of -0.0755.

Me personally I was neither suprised nor expected Regent Sqaure to be the winner as im not form the Pittsburgh area, 
but from what ive reseached after doing this project it seems like a desirable place to live.